overlord/devicestate/devicestate.go

```diff
@@ -508,10 +509,28 @@ func remodelTasks(ctx context.Context, st *state.State, current, new *asserts.Mo
 		// create a recovery when remodeling to a UC20 system, actual
 		// policy for possible remodels has already been verified by the
 		// caller
-		label := timeNow().Format("20060102")
-		createRecoveryTasks, err := createRecoverySystemTasks(st, label, snapSetupTasks)
-		if err != nil {
-			return nil, err
+		labelBase := timeNow().Format("20060102")
+		labelSuffix := ""
+		var createRecoveryTasks *state.TaskSet
+		// the label is based on date, but if multiple remodels are done
+		// on the same day, we may end up with conflicting labels
+		const maxLabelTries = 5
```

Comment:
given that minimal model changes might reuse the same snaps, it unclear how many recovery systems one can stick in the seed even it's not very large.  My preference is to try "<base-label>-1", if that doesn't exist use it, if it exists glob "<base-label>-*", find the max for those where the suffix can be turned into a number, and add one